### PR TITLE
icedtea-web: bump to 1.8.3

### PR DESCRIPTION
**Pull request by alarcher**

### CODE REVIEW

File: components/web/icedtea-web/firefox-java.p5m [hunk 1]
```diff
@@ -22,19 +22,36 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
...
-depend type=require fmri=pkg:/runtime/java/openjdk7
+depend type=require fmri=pkg:/runtime/java/openjdk8
 
 # Generated section:
+file path=usr/bin/itw-modularjdk.args
 file path=usr/bin/itweb-settings
 file path=usr/bin/javaws
 file path=usr/bin/policyeditor
-file path=usr/lib/IcedTeaPlugin.so mode=0555
+file path=usr/lib/$(MACH64)/IcedTeaPlugin.so
```
Comment (pyhalov) [inline components/web/icedtea-web/firefox-java.p5m]: Do I understand correctly that only user interface of icedtea is javaws, and IcedTeaPlugin can't be used by browser? So why do we ship it?

File: components/web/icedtea-web/Makefile [hunk 2]
```diff
@@ -13,40 +13,38 @@
...
-   http://icedtea.wildebeest.org/download/source/$(COMPONENT_ARCHIVE)
+   https://github.com/AdoptOpenJDK/IcedTea-Web/archive/$(COMPONENT_ARCHIVE)
 COMPONENT_PROJECT_URL=	http://icedtea.classpath.org/wiki/Main_Page
 COMPONENT_FMRI=	web/browser/firefox/plugin/firefox-java
 COMPONENT_CLASSIFICATION=	Applications/Plug-ins and Run-times
 COMPONENT_LICENSE=	GPLv2
 
-#Use Java 7
-JAVA_HOME=$(JAVA7_HOME)
+include $(WS_MAKE_RULES)/common.mk
```
Comment (Mno-hime) [inline components/web/icedtea-web/Makefile:35]: `TEST_TARGET=$(NO_TESTS)` ?

**Comment by 3eka** [comment]
Hi Alp,
I would like to see this in distribution (if not big deal).
Sometimes some server consoles use jnlp extension.

Regards.

**Comment by pyhalov** [comment]
I understand this, but does javaws need IcedTeaPlugin.so?

**Comment by 3eka** [comment]
Heh, not sure. This is what has confused me:
```
:; pkg search /usr/bin/javaws
INDEX      ACTION VALUE          PACKAGE
path       file   usr/bin/javaws pkg:/web/browser/firefox/plugin/firefox-java@1.5.3-2018.0.0.1
```
and several lines from that shell file are:
```
grep -r icedtea /usr/bin/javaws
LAUNCHER_BOOTCLASSPATH=-Xbootclasspath/a:/usr/share/icedtea-web/netx.jar:/usr/jdk/instances/openjdk1.7.0/jre/lib/rhino.jar:
SPLASH_LOCATION=/usr/share/icedtea-web/javaws_splash.png
CUSTOM_JRE=`grep "$CUSTOM_JRE_REGEX" $CONFIG_HOME/icedtea-web/deployment.properties 2>/dev/null |  sed "s/$CUSTOM_JRE_REGEX//g"`
  CUSTOM_JRE=`grep "$CUSTOM_JRE_REGEX" ~/.icedtea/deployment.properties 2>/dev/null |  sed "s/$CUSTOM_JRE_REGEX//g"`
COMMAND[k]="-Dicedtea-web.bin.name=${PROGRAM_NAME}"
COMMAND[k]="-Dicedtea-web.bin.location=${BINARY_LOCATION}"
```
Regards.

**Comment by alarcher** [comment]
@pyhalov @Mno-hime  I have no clue what this component is used for :) feel free to remove files that are not required.

**Comment by Mno-hime** [comment]
The updated version keeps asking this:
```
$ javaws typeahead.jnlp 
selected jre: /usr/jdk/instances/openjdk1.8.0/jre/
Got 'Keystore was tampered with, or password was incorrect'  during keystore operation 'load keystore'. Attempts to unlock: 1
Invalid password?
```

**Comment by pyhalov** [comment]
icedtea needs rust to compile

File: components/web/icedtea-web/Makefile (hunk 2 above)
Comment (alarcher) [reply components/web/icedtea-web/Makefile:35]: Fixed

File: components/web/icedtea-web/firefox-java.p5m (hunk 1 above)
Comment (alarcher) [reply components/web/icedtea-web/firefox-java.p5m]: Removed

**Comment by alarcher** [comment]
> icedtea needs rust to compile

Fixed

**Comment by alarcher** [comment]
@pyhalov Feel free to squash and merge if it works for you.

**Comment by pyhalov** [comment]
@alarcher  Please, create separate PR with OpenJDK keystore password change.

**Pull request merged**